pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -468,6 +470,23 @@ func (ds *Plugin) CreateFederationRelationship(ctx context.Context, fr *datastor
 	})
 }
 
+// FetchFederationRelationship fetches the federation relationship that matches
+// the given trust domain. If the federation relationship is not found, nil is returned.
+func (ds *Plugin) FetchFederationRelationship(ctx context.Context, trustDomain spiffeid.TrustDomain) (fr *datastore.FederationRelationship, err error) {
+	if trustDomain.IsZero() {
+		return nil, status.Errorf(codes.InvalidArgument, "trust domain is required")
```

Comment:
nit: no need for formatted error
```suggestion
		return nil, status.Error(codes.InvalidArgument, "trust domain is required")
```